server/backend/db_backend.go

```diff
@@ -554,27 +584,60 @@ func (self *MongoBackend) getTransactionList(
 	filter *models.TxsFilter,
 ) ([]*models.Transaction, error) {
 	var transactions []*models.Transaction
-	findQuery := bson.M{
-		"$or": []bson.M{
-			{"from": address},
-			{"to": address},
-		},
-		"created_at": bson.M{
-			"$gte": filter.FromTime,
-			"$lte": filter.ToTime,
-		},
-	}
-	if filter.InputDataEmpty != nil {
-		findQuery["input_data_empty"] = *filter.InputDataEmpty
-	}
-	err := self.mongo.C("Transactions").
-		Find(findQuery).
-		Sort("-created_at").
-		Skip(filter.Skip).
-		Limit(filter.Limit).
-		All(&transactions)
-	if err != nil {
-		return nil, fmt.Errorf("failed to get tx list: %v", err)
+	if self.useTransactionsByAddress() {
+		var transactionsList []*models.TransactionsByAddressAggregated
+		findQuery := bson.M{
+			"address": address,
+			"created_at": bson.M{
+				"$gte": filter.FromTime,
+				"$lte": filter.ToTime,
+			},
+		}
+		query := []bson.M{
+			{"$match": findQuery},
+			{"$lookup": bson.M{
+				"from":         "Transactions",
+				"localField":   "tx_hash",
+				"foreignField": "tx_hash",
+				"as":           "tx",
+			}},
+			{"$unwind": bson.M{
+				"path": "$tx",
+			}},
```

Comment:
Could we add something like `{ $replaceRoot: { newRoot: "$tx" } }` here? Then we can drop the extra types and just use `models.Transaction` directly.